vortex-array/src/compute/is_constant.rs

```diff
@@ -136,17 +136,19 @@ fn is_constant_impl(
     }
 
     // We already know here that the array is all valid, so we check for min/max stats.
-    let min = array
-        .statistics()
-        .get_scalar(Stat::Min, array.dtype())
-        .and_then(|p| p.as_exact());
-    let max = array
-        .statistics()
-        .get_scalar(Stat::Max, array.dtype())
-        .and_then(|p| p.as_exact());
+    let min = array.statistics().get_scalar(Stat::Min, array.dtype());
+    let max = array.statistics().get_scalar(Stat::Max, array.dtype());
 
     if let Some((min, max)) = min.zip(max) {
-        if min == max {
+        // min/max are equal and exact and there are no NaNs
+        if min.is_exact()
+            && min == max
+            && if array.dtype().is_float() {
+                array.statistics().get_as::<u64>(Stat::NaNCount) == Some(Precision::exact(0u64))
```

Comment:
nan_count(array)? == 0? You still have to compute the nan_count otherwise you might get a false positive still?

Reply:
I think the question is: if you're this deep into the short-circuit call (min max are present, and equal), then you may as well perform actual nan_count compute instead of also requiring that stat to be pre-computed.

Reply:
I think the overall thing is that it's faster to fallback to is_constant than doing nan_count here. Is there a nicer way of expressing this than an if withing an if condition?

Reply:
Is there something you want to change here?